src/App.vue

```diff
@@ -105,6 +105,10 @@ export default {
   async beforeMount() {
     await this.appLoad();
   },
+  created() {
+    // will trigger function before closing/refreshing tab
+    window.addEventListener('beforeunload', this.showConfirmationPrompt);
```

Comment:
You have to remove the eventlistenner in the vuejs event `beforeDestroy`

Reply:
right, done

Reply:
You've put it in `mounted`. Check the lifecycle of vuejs: https://vuejs.org/v2/guide/instance.html#Instance-Lifecycle-Hooks

Reply:
The removeListener is inside `beforeDestroy`, please expand the code above

Reply:
My bad.

Reply:
no problem